src/bare_item.rs

```diff
@@ -0,0 +1,626 @@
+        Self::new_decimal(item)
+    }
+}
+
+impl TryFrom<f64> for BareItem {
+    type Error = &'static str;
+
+    /// Converts `f64` into `BareItem::Decimal`.
+    /// ```
+    /// # use sfv::{BareItem, FromPrimitive};
+    /// # use std::convert::TryInto;
+    /// # use rust_decimal::prelude::ToPrimitive;
+    /// # fn main() -> Result<(), &'static str> {
+    /// let decimal_number = 48.01;
+    /// let bare_item: BareItem = decimal_number.try_into()?;
+    /// assert_eq!(bare_item.as_decimal().unwrap().to_f64().unwrap(), decimal_number);
+    /// # Ok(())
+    /// # }
+    /// ```
+    fn try_from(value: f64) -> Result<Self, Self::Error> {
+        Self::new_decimal_from_f64(value)
+    }
+}
+
+impl TryFrom<&[u8]> for BareItem {
+    type Error = &'static str;
+
+    /// Converts a byte slice into `BareItem::ByteSeq`.
+    /// ```
+    /// # use sfv::{BareItem, FromPrimitive};
+    /// # use std::convert::TryInto;
+    /// # fn main() -> Result<(), &'static str> {
+    /// let byte_slice = "hello".as_bytes();
+    /// let bare_item: BareItem = byte_slice.try_into()?;
+    /// assert_eq!(bare_item.as_byte_seq().unwrap(), byte_slice);
+    /// # Ok(())
+    /// # }
+    /// ```
+    fn try_from(value: &[u8]) -> Result<Self, Self::Error> {
+        Self::new_byte_seq(value)
+    }
+}
+
+impl TryFrom<bool> for BareItem {
+    type Error = &'static str;
+
+    /// Converts a `bool` into `BareItem::Boolean`.
+    /// ```
+    /// # use sfv::{BareItem, FromPrimitive};
+    /// # use std::convert::TryInto;
+    /// # fn main() -> Result<(), &'static str> {
+    /// let boolean = true;
+    /// let bare_item: BareItem = boolean.try_into()?;
+    /// assert_eq!(bare_item.as_bool().unwrap(), boolean);
+    /// # Ok(())
+    /// # }
+    /// ```
+    fn try_from(value: bool) -> Result<Self, Self::Error> {
+        Self::new_boolean(value)
+    }
+}
+
+/// Decimals are numbers with an integer and a fractional component. The integer component has at most 12 digits; the fractional component has at most three digits.
+///
+/// The ABNF for decimals is:
+/// ```abnf,ignore,no_run
+/// sf-decimal  = ["-"] 1*12DIGIT "." 1*3DIGIT
+/// ```
+#[derive(Debug, PartialEq, Clone)]
+pub struct BareItemDecimal(pub(crate) rust_decimal::Decimal);
+
+impl TryFrom<rust_decimal::Decimal> for BareItemDecimal {
+    type Error = &'static str;
+    fn try_from(value: rust_decimal::Decimal) -> Result<Self, Self::Error> {
+        let validated = Self::validate(value)?;
+        Ok(BareItemDecimal(validated))
+    }
+}
+
+impl ValidateValue<'_, rust_decimal::Decimal> for BareItemDecimal {
+    fn validate(value: rust_decimal::Decimal) -> SFVResult<rust_decimal::Decimal> {
+        let fraction_length = 3;
+
+        let decimal = value.round_dp(fraction_length);
+        let int_comp = decimal.trunc();
+        let int_comp = int_comp
+            .abs()
+            .to_u64()
+            .ok_or("serialize_decimal: integer component > 12 digits")?;
```

Comment:
@valenting is it acceptable to rename those errors all to `validate_*` rather than `serialize_*`? 

Those errors can now occur during value creation _and_ during serialization. With the `&'static str` error type I couldn't find a way to "prefix" the value with `serialize_` or `validate_` depending on where it originated. 

Any input is appreciated 🙂